### PR TITLE
stats: statistics on request and response body size

### PR DESCRIPTION
**Pull request by zyfjeff**
Signed-off-by: tianqian.zyf <tianqian.zyf@alibaba-inc.com>

*Description*: statistics on request and response body size
*Risk Level*: low
*Testing*: unittest 
*Docs Changes*: yes
*Release Notes*: N/A
[Optional Fixes #Issue] #3621 
[Optional *Deprecated*:]

### CODE REVIEW

File: docs/root/configuration/cluster_manager/cluster_stats.rst [hunk 1]
```diff
@@ -50,8 +50,10 @@ Every cluster has a statistics tree rooted at *cluster.<name>.* with the followi
   upstream_cx_destroy_local_with_active_rq, Counter, Total connections destroyed locally with 1+ active request
   upstream_cx_destroy_remote_with_active_rq, Counter, Total connections destroyed remotely with 1+ active request
   upstream_cx_close_notify, Counter, Total connections closed via HTTP/1.1 connection close header or HTTP/2 GOAWAY
+  upstream_cx_rx_bytes_total, Histogram, Size of received connection bytes
```
[the remark this answers is not in the document]
Comment (zyfjeff) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: Thank you, misspelled

File: docs/root/configuration/cluster_manager/cluster_stats.rst [hunk 2]
```diff
@@ -50,8 +50,10 @@ Every cluster has a statistics tree rooted at *cluster.<name>.* with the followi
   upstream_cx_destroy_local_with_active_rq, Counter, Total connections destroyed locally with 1+ active request
   upstream_cx_destroy_remote_with_active_rq, Counter, Total connections destroyed remotely with 1+ active request
   upstream_cx_close_notify, Counter, Total connections closed via HTTP/1.1 connection close header or HTTP/2 GOAWAY
+  upstream_cx_rx_bytes, Histogram, Size of received connection bytes
   upstream_cx_rx_bytes_total, Counter, Total received connection bytes
   upstream_cx_rx_bytes_buffered, Gauge, Received connection bytes currently buffered
+  upstream_cx_tx_bytes_total, Histogram, Size of sent connection bytes
```
Comment (dnoe) [inline docs/root/configuration/cluster_manager/cluster_stats.rst]: I think it needs to be fixed here too.

Comment (zyfjeff) [reply docs/root/configuration/cluster_manager/cluster_stats.rst]: sorry, I check again carefully

**Comment by dnoe** [comment]
Assigning @zuercher for senior review.

File: docs/root/configuration/network_filters/tcp_proxy_filter.rst [hunk 3]
```diff
@@ -22,8 +22,10 @@ statistics are rooted at *tcp.<stat_prefix>.* with the following statistics:
 
   downstream_cx_total, Counter, Total number of connections handled by the filter
   downstream_cx_no_route, Counter, Number of connections for which no matching route was found or the cluster for the route was not found
+  downstream_cx_tx_bytes, Histogram, Bytes written to the downstream connection
```
Comment (zuercher) [inline docs/root/configuration/network_filters/tcp_proxy_filter.rst]: Nit: Should say "Size of bytes written" (and "Size of bytes read" below) to match the others.

**Comment by zuercher** [comment]
@zyfjeff are you still working on this?

**Comment by zuercher** [comment]
@zyfjeff I apologize, but I don't understand what you're asking. Could you restate it?

In the meantime, I can give you some information that might help. The HTTP connection manager/router already track the number of bytes received and sent at the request level (it's stored in RequestInfo). The HTTP router also tracks the bytes received and sent for each proxied upstream request in a separate RequestInfo. Perhaps it's sufficient to expose those as stats? (We'd probably want to ask the original issue author.) Request body size tracking could be done in a similar way.

**Comment by zyfjeff** [comment]
At present the implementation of the knowledge of each socket read data size, is not really the size of a request, does not meet the demand, so the close off the PR.

**Pull request closed without merging**